_posts/2018-11-12-cloud-cloud-computing.md

```diff
@@ -0,0 +1,89 @@
+Virtualization software, storage for your application and related libraries, operating system and networking are all provided, in the cloud. 
+
+
+You will, however, have to manage your own operating system, provide middleware example an antivirus, provide your application with data and management system for it, if your application needs a database you would have to create it, provide runtime libraries and the actual application.
+
+
+
+### PaaS
+
+Paas stands for Platform as a Service. The service provider provides you with a pre-configured infrastructure.PaaS includes middleware, runtimes, development, testing and deployment tools.
+Here's a picture;
+
+![PaaS model](/assets/images/blog/cloud-computing/paas.png){:class="img-responsive center"}
+
+
+
+All you provide is code for your application and the data.
+Example of Paas providers are [Heroku](https://www.heroku.com/), [Go daddy](https://uk.godaddy.com), etc
+
+In this case, all you will have to do is create code for your application, then upload the files for your provider to deploy.
+
+
+### SaaS
+Software as a service. The service provider fully manages the whole stack. It is an application hosted in a public domain therefore accessible to everyone.
+Here's a picture;
+
+![SaaS model](/assets/images/blog/cloud-computing/saas.png){:class="img-responsive center"}
+
+The only thing you manage is your own space. Your own environment where you have your own data.
+
+A good example of a Software as a service is Gmail provided by Google.
+
+
+## Essential characteristics
+
+
+1. **On-demand self-service.** A consumer can provision cloud resources on demand, such as the server, time and network storage, as needed automatically without requiring human interaction with each service provider. whenever they are required.
+
+2. **Broad Network access.** Resources are available for access from a wide range of devices, such as tablets, PCs, Macs, and smartphones. These resources are also accessible from a wide range of locations that offer online access.
+
+3. **Resource pooling.** Different physical and virtual resources dynamically assigned and reassigned according to consumer demand. There is a sense of location independence in that the customer generally has no control or knowledge over the exact location of the provided resources but may be able to specify location at a higher level of abstraction (e.g country, state, or datacenter). 
+
+4. **Rapid elasticity.** Scalable provisioning, or the ability to provide scalable services. To the consumer, the capabilities available for provisioning often appear to be unlimited and can be appropriated in any quantity at any time. 
+
+5. **Measured service.** Usage can be monitored, controlled, and reported, providing transparency for both the provider and consumer of the utilized service. 
```

Comment:
Need to add developing and testing cloud infrastructure on your local host. For IaaS, you'd look at vagrant, virtual box, vmware etc. For Kubernetes you'd look at rancher etc and for PaaS you'd look at flynn, dokku, deis.

Reply:
Please add this as well.